docker/docker.go

```diff
@@ -571,5 +584,71 @@ func Attach(ctx context.Context, config *container.Config, host *container.HostC
 		exit <- code
 	}()
 
+	monitorTtySize(c, res.ID)
+
 	return &resp, exit, nil
 }
+
+func getStdOutSize() (uint, uint) {
+	fd, isTerminal := term.GetFdInfo(os.Stdout)
+	if !isTerminal {
+		return 0, 0
+	}
+
+	ws, err := term.GetWinsize(fd)
+	if err != nil {
+		return 0, 0
+	}
+
+	return uint(ws.Height), uint(ws.Width)
+}
+
+func monitorTtySize(c *client.Client, containerID string) {
+	initTtySize(c, containerID)
+	if runtime.GOOS == "windows" {
+		go func() {
+			prevH, prevW := getStdOutSize()
+			for {
+				time.Sleep(time.Millisecond * 250)
+				h, w := getStdOutSize()
+
+				if prevW != w || prevH != h {
+					resizeTty(c, containerID)
+				}
+				prevH = h
+				prevW = w
+			}
+		}()
+	} else {
+		sigchan := make(chan os.Signal, 1)
+		gosignal.Notify(sigchan, signal.SIGWINCH)
+		go func() {
+			for range sigchan {
+				resizeTty(c, containerID)
+			}
+		}()
+	}
+}
+
+// initTtySize is to init the tty's size to the same as the window, if there is an error, it will retry 5 times.
+func initTtySize(c *client.Client, containerID string) {
+	if err := resizeTty(c, containerID); err != nil {
+		go func() {
+			var err error
+			for retry := 0; retry < 5; retry++ {
+				time.Sleep(10 * time.Millisecond)
+				if err = resizeTty(c, containerID); err == nil {
+					break
+				}
+			}
+		}()
+	}
+}
+
+func resizeTty(c *client.Client, containerID string) error {
```

Comment:
Maybe better to change this into
```
func resizeTty(c *client.Client, containerID string, width, height uint) error
```
so that in case of windows you don't calculate `getStdOutSize` twice?

Reply:
No strong opinion on it. I just think it makes code easier. The overhead is super small here. Like REALLY small.
Please let me know if you think that additional overhead is more important than code simplicity. I'll change it.

Reply:
Ok, I was actually concerned about the overhead 👍